internal/postrs/proof.go

```diff
@@ -34,7 +35,26 @@ func TranslateScryptParams(n, r, p uint) ScryptParams {
 	}
 }
 
-func GenerateProof(dataDir string, challenge []byte, logger *zap.Logger, nonces, threads uint, K1, K2 uint32, powDifficulty [32]byte, powFlags PowFlags) (*shared.Proof, error) {
+type postOptions struct {
+	powCreatorId []byte
+}
+
+type PostOptionFunc func(*postOptions) error
+
+func WithPowCreator(id []byte) PostOptionFunc {
```

Comment:
Depending on the effort, would it make sense to change all the other properties also to functional options?

Reply:
Created issue to address later: https://github.com/spacemeshos/post/issues/174